File: src/vovka_pkg/scripts/producer.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import String


def producer():
    publisher = rospy.Publisher('market', String, queue_size=5)
    rospy.init_node('producer', anonymous=True)
    rate = rospy.Rate(2)
    
    brands = ['Samsung', 'Lenovo', 'LG']
    series = ['I', 'Q', 'J', 'S', 'C']
    count = 0
    while not rospy.is_shutdown():
        msg = 'Just like origin %s %s%d' % (brands[count % len(brands)], series[count % len(series)], (count % len(brands) * len(series)))
        rospy.loginfo(msg)
        publisher.publish(msg)

        count += 1
        rate.sleep()


if __name__ == '__main__':
    try:
        producer()
    except rospy.ROSInterruptExcetion:
        pass

File: src/vovka_pkg/src/producer.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"

#include <sstream>
#include <vector>
#include <string>

int main(int argc, char **argv) {
    
    ros::init(argc, argv, "producer");
    ros::NodeHandle node;

    ros::Publisher publisher = node.advertise<std_msgs::String>("market", 5);
    ros::Rate loop_rate(2);

    int count = 0;
    std::vector<std::string> brands;
    brands.push_back("IPhone");
    brands.push_back("Samsung");
    brands.push_back("Nokia");
    std::vector<std::string> series;
    series.push_back("S"); 
    series.push_back("A"); 
    series.push_back("K"); 
    series.push_back("J"); 
    series.push_back("C"); 

    while(ros::ok()) {
        std_msgs::String msg;

        std::stringstream ss;
        ss << "Brand new " << brands[count % brands.size()] << " " << series[count % series.size()] << count / (brands.size() * series.size());

        msg.data = ss.str();
        ROS_INFO("%s", msg.data.c_str());

        publisher.publish(msg);
        ros::spinOnce();

        loop_rate.sleep();
        ++count;
    }

    return 0;
}
